packages/docgen/lib/get-type-annotation.js

```diff
@@ -0,0 +1,424 @@
+		if ( ! qualifiedName.left ) {
+			return qualifiedName.name;
+		}
+		return qualifiedName.left.name;
+	}
+	return `${ unifyQualifiedName( qualifiedName.left ) }.${
+		qualifiedName.right.name
+	}`;
+}
+
+/**
+ * @param {babelTypes.TSImportType} typeAnnotation
+ */
+function getImportTypeAnnotation( typeAnnotation ) {
+	// Should this just return the unqualified name (i.e., typeAnnotation.name || typeAnnotation.right.name)?
+	return `import('${ typeAnnotation.argument.value }').${ unifyQualifiedName(
+		typeAnnotation.qualifier
+	) }`;
+}
+
+/**
+ *
+ * @param {babelTypes.TSType} objectType
+ */
+function getIndexedAccessTypeAnnotationObjectName( objectType ) {
+	if ( babelTypes.isTSImportType( objectType ) ) {
+		return objectType.qualifier.name;
+	}
+	return objectType.typeName.name;
+}
+
+/**
+ * @param {babelTypes.TSIndexedAccessType} typeAnnotation
+ */
+function getIndexedAccessTypeAnnotation( typeAnnotation ) {
+	const objName = getIndexedAccessTypeAnnotationObjectName(
+		typeAnnotation.objectType
+	);
+	const index = typeAnnotation.indexType.literal.value;
+	return `${ objName }[ '${ index }' ]`;
+}
+
+/**
+ *
+ * @param {babelTypes.TSLiteralType} typeAnnotation
+ */
+function getLiteralTypeAnnotation( typeAnnotation ) {
+	switch ( typeAnnotation.literal.type ) {
+		case 'BigIntLiteral': {
+			return `${ typeAnnotation.literal.value }n`;
+		}
+		case 'NumericLiteral':
+		case 'BooleanLiteral': {
+			return typeAnnotation.literal.value;
+		}
+		case 'StringLiteral': {
+			return `'${ typeAnnotation.literal.value }'`;
+		}
+	}
+}
+
+/**
+ * @param {babelTypes.TSMappedType} typeAnnotation
+ */
+function getMappedTypeAnnotation( typeAnnotation ) {
+	const typeParam = typeAnnotation.typeParameter.name;
+	const constraintOperator = typeAnnotation.typeParameter.constraint.operator;
+	const constraintAnnotation = getTypeAnnotation(
+		typeAnnotation.typeParameter.constraint.typeAnnotation
+	);
+	const mappedValue = getTypeAnnotation( typeAnnotation.typeAnnotation );
+	return `[ ${ typeParam } in ${ constraintOperator } ${ constraintAnnotation } ]: ${ mappedValue }`;
+}
+
+/**
+ * @param {babelTypes.TSTypeReference} typeAnnotation
+ */
+function getTypeReferenceTypeAnnotation( typeAnnotation ) {
+	if ( ! typeAnnotation.typeParameters ) {
+		return typeAnnotation.typeName.name;
+	}
+	const typeParams = typeAnnotation.typeParameters.params
+		.map( getTypeAnnotation )
+		.join( ', ' );
+	return `${ typeAnnotation.typeName.name }< ${ typeParams } >`;
+}
+
+/**
+ * @param {TypeAnnotation} typeAnnotation
+ * @return {string | null} The type or null if not an identifiable type.
+ */
+function getTypeAnnotation( typeAnnotation ) {
+	switch ( typeAnnotation.type ) {
+		case 'TSAnyKeyword': {
+			return 'any';
+		}
+		case 'TSArrayType': {
+			return getArrayTypeAnnotation( typeAnnotation );
+		}
+		case 'TSBigIntKeyword': {
+			return 'BigInt';
+		}
+		case 'TSBooleanKeyword': {
+			return 'boolean';
+		}
+		case 'TSConditionalType': {
```

Comment:
Couldn't figure out how to make one of these. If anyone knows, would be much appreciated 😃